src/ConnectivityWizard/Endpoints/ScanEndpoint.ts

```diff
@@ -0,0 +1,194 @@
+import { Endpoint } from './Endpoint';
+import { ScannedUrl } from '../Entities/ScannedUrl';
+import { Tokens } from '../Entities/Tokens';
+import { ScanId } from '../Entities/ScanId';
+import { TokensOperations } from '../TokensOperations';
+import logger from '../../Utils/Logger';
+import Koa from 'koa';
+import child_processes from 'child_process';
+
+export class ScanEndpoint implements Endpoint {
+  private tokenOperations: TokensOperations;
+  private repeaterProcess: child_processes.ChildProcess;
+
+  constructor(tokenOps: TokensOperations) {
+    this.tokenOperations = tokenOps;
+  }
+
+  public async handle(ctx: Koa.Context): Promise<void> {
+    const req = <ScannedUrl>ctx.request.body;
+    const tokens: Tokens = this.tokenOperations.readTokens();
+    let scan_id: string = null;
+
+    if (!this.executeRepeater(tokens)) {
+      ctx.throw('Could not execute Repeater');
+
+      return;
+    }
+
+    try {
+      scan_id = await this.launchScan(req.url, tokens);
+    } catch (err) {
+      ctx.status = 400;
+
+      return;
+    }
+
+    ctx.body = <ScanId>{
+      scanId: scan_id
+    };
+  }
+
+  private executeRepeater(tokens: Tokens): boolean {
+    if (this.repeaterProcess != null && this.repeaterProcess.exitCode == null) {
+      logger.debug(
+        `Repeater process is still runnning (PID ${this.repeaterProcess.pid}). Skippiing spawning another one`
+      );
+
+      return true;
+    }
+
+    const node_exec = this.getNodeExec();
+
+    try {
+      const startArgs: string[] = [
+        ...node_exec.argv,
+        'repeater',
+        '--token',
+        `"${tokens.authToken}"`,
+        '--agent',
+        `"${tokens.repeaterId}`
+      ];
+      logger.debug(
+        `Launching Repeater process with cmd: ${
+          node_exec.cmd
+        } and arguments: ${JSON.stringify(startArgs)}`
+      );
+
+      this.repeaterProcess = child_processes.spawn(node_exec.cmd, startArgs, {
+        detached: true
+      });
+
+      this.repeaterProcess.stdout.on('data', (data) => {
+        const line = data.toString();
+        logger.debug(`Repeater (stdout): ${line}`);
+      });
+      this.repeaterProcess.stderr.on('data', (data) => {
+        const line = data.toString();
+        logger.error(`Repeater (stderr): ${line}`);
+      });
+
+      this.repeaterProcess.unref();
+
+      let fired = false;
```

Comment:
You don't need to control output. I don't see any issue if we get 2 messages: error message and code of exit. As I said early, `exit` and `close` can't happen at the same time.

Reply:
There are cases both events fire